src/DynamoPackagesUI/ViewModels/PackageManagerViewModel.cs

```diff
@@ -18,28 +27,533 @@ namespace Dynamo.DynamoPackagesUI.ViewModels
+                Resources.PackageDownloadConfirmMessageBoxTitle,
+                MessageBoxButton.OKCancel, MessageBoxImage.Question, asset["asset_name"], version["version"]) :
+                PkgMgrCommands.Show(PackageManagerMessages.CONFIRM_TO_INSTALL_PACKAGE_FOLDER,
+                Resources.PackageDownloadConfirmMessageBoxTitle,
+                MessageBoxButton.OKCancel, MessageBoxImage.Question, asset["asset_name"], version["version"], downloadPath);
+
+            if (PackagesToInstall == null)
+                PackagesToInstall = new List<string>();
+            else
+                PackagesToInstall.Clear();
+
+            if (result == MessageBoxResult.OK)
+            {
+                if (!string.IsNullOrEmpty(version["dependencies"]))
+                {
+                    GetDependencies(version, out packageVersionData);
+                }
+
+                //Add selected package to the list of packges to install
+                PackagesToInstall.Add(string.Format("{0},{1},{2}", asset["asset_id"], version["file_id"], asset["asset_name"]));
+
+                // determine if any of the packages contain binaries or python scripts.  
+                result = CheckForBinariesPythonScripts(version, packageVersionData);
+                if (result == MessageBoxResult.Cancel)
+                    return "cancel";
+
+                result = CheckForNewerDynamoVersion(packageVersionData);
+                if (result == MessageBoxResult.Cancel)
+                    return "cancel";
+
+
+                var localPkgs = PkgMgrCommands.LocalPackages;
+
+                var uninstallsRequiringRestart = new List<Package>();
+                var uninstallRequiringUserModifications = new List<Package>();
+                var immediateUninstalls = new List<Package>();
+
+                // if a package is already installed we need to uninstall it, allowing
+                // the user to cancel if they do not want to uninstall the package
+                foreach (var localPkg in packageVersionData.Select(x => localPkgs.FirstOrDefault(v => v.Name == x.Item1.asset_name.ToString())))
+                {
+                    if (localPkg == null) continue;
+
+                    if (localPkg.LoadedAssemblies.Any())
+                    {
+                        uninstallsRequiringRestart.Add(localPkg);
+                        continue;
+                    }
+
+                    if (localPkg.InUse(Model))
+                    {
+                        uninstallRequiringUserModifications.Add(localPkg);
+                        continue;
+                    }
+
+                    immediateUninstalls.Add(localPkg);
+                }
+
+                if (uninstallRequiringUserModifications.Any())
+                {
+                    PkgMgrCommands.Show(PackageManagerMessages.UNINSTALL_TO_CONTINUE, Resources.CannotDownloadPackageMessageBoxTitle,
+                        MessageBoxButton.OK, MessageBoxImage.Error, ProductName,
```

Comment:
looks like you need to construct the message string locally and pass it to show method.